src/repos/FlyerRepo.ts

```diff
@@ -0,0 +1,229 @@
+      })
+  );
+};
+
+/**
+ * Performs fuzzy search on all Flyers to find Flyers with title/publisher matching the query.
+ * @param query the term to search for
+ * @param limit the number of results to return
+ * @returns at most limit Flyers with titles or publishers matching the query
+ */
+const searchFlyers = async (query: string, limit = DEFAULT_LIMIT) => {
+  const allFlyers = await FlyerModel.find({});
+  const searcher = new Fuse(allFlyers, {
+    keys: ['title', 'organization.name'],
+  });
+
+  return searcher
+    .search(query)
+    .map((searchRes) => searchRes.item)
+    .slice(0, limit);
+};
+
+/**
+ * Computes and returns the trending Flyers in the database.
+ *
+ * @function
+ * @param {number} limit - number of Flyers to retrieve.
+ */
+const getTrendingFlyers = async (limit = DEFAULT_LIMIT): Promise<Flyer[]> => {
+  const flyers = await FlyerModel.find({ isTrending: true }).exec();
+  return flyers.filter((flyer) => !isFlyerFiltered(flyer)).slice(0, limit);
+};
+
+/**
+ * Refreshes trending Flyers.
+ */
+const refreshTrendingFlyers = async (): Promise<Flyer[]> => {
+  // Set previous trending Flyers to not trending
+  const oldTrendingFlyers = await FlyerModel.find({ isTrending: true }).exec();
+  oldTrendingFlyers.forEach(async (a) => {
+    const flyer = await FlyerModel.findById(new ObjectId(a._id)); // eslint-disable-line
+    flyer.isTrending = false;
+    await flyer.save();
+  });
+
+  // Get new trending Flyers
+  const flyers = await FlyerModel.aggregate()
+    // Get a sample of random Flyers
+    .sample(100)
+    // Get Flyers after 30 days ago
+    .match({
+      date: {
+        $gte: new Date(
+          new Date().setDate(new Date().getDate() - MAX_NUM_DAYS_OF_TRENDING_ARTICLES),
+        ),
+      },
+    });
+
+  flyers.forEach(async (a) => {
+    const flyer = await FlyerModel.findById(new ObjectId(a._id)); // eslint-disable-line
+    flyer.isTrending = true;
+    await flyer.save();
+  });
+
+  return flyers;
+};
+
+/**
+ * Increments number of shoutouts on an Flyer and publication by one.
+ * @function
+ * @param {string} id - string representing the unique Object Id of an Flyer.
+ */
+const incrementShoutouts = async (id: string): Promise<Flyer> => {
+  const flyer = await FlyerModel.findById(new ObjectId(id));
+  if (flyer) {
+    flyer.shoutouts += 1;
+    return flyer.save();
+  }
+  return flyer;
+};
+
+/**
+ * Checks if an Flyer's title contains profanity.
+ * @function
+ * @param {string} title - Flyer title.
+ */
+const checkProfanity = async (title: string): Promise<boolean> => {
+  const filter = new Filter();
+  return filter.isProfane(title);
+};
+
+export default {
+  checkProfanity,
+  getAllFlyers,
+  getFlyerByID,
+  getFlyersAfterDate,
+  getFlyersByIDs,
+  getFlyersByOrganizationID,
+  getFlyersByOrganizationIDs,
+  getFlyersByOrganizationSlug,
+  getFlyersByOrganizationSlugs,
+  searchFlyers,
+  getTrendingFlyers,
+  incrementShoutouts,
+  refreshTrendingFlyers,
```

Comment:
alphabetize

Reply:
appdev teaching me the alphabet im so thankful